src/ContentChanges.ts

```diff
@@ -54,19 +52,27 @@ export class ContentChanges {
   }
 
   public getChanges(): ChangeList {
-    return this.changes
+    return Array.from(this.changes.values())
   }
 
   public isEmpty(): boolean {
-    return this.changes.length === 0
+    return this.changes.size === 0
+  }
+
+  private add(address: SimpleCellAddress, change: CellValueChange) {
+    const value = change.value
+    if (value instanceof SimpleRangeValue) {
+      for (const cellAddress of value.effectiveAddressesFromData(address)) {
+        this.changes.delete(`${cellAddress.sheet},${cellAddress.col},${cellAddress.row}`)
+      }
+    }
+    this.changes.set(addressKey((address)), change)
```

Comment:
I have not seen how it is used, but: what if first `SimpleRangeValue` is inserted and later a scalar value?
Or maybe this scenario does not happen?

Reply:
This scenario should not happen. It can be handled nicer but at this moment there are to much conflicts in places not related to this PR.